src/watchmaker/__init__.py

```diff
@@ -301,6 +304,12 @@ def _get_config(self):
             self.log.critical(msg)
             raise WatchmakerException(msg)
 
+        if config_version_specifier and not check_version(
+                watchmaker.__version__, config_version_specifier):
+            msg = 'Watchmaker is not compatible with the config file'
```

Comment:
I think it would be helpful to output the version and the constraint in the msg...
```suggestion
            msg = 'Watchmaker version {} is not compatible with the config file (watchmaker_version = {})'.format(watchmaker.__version__, config_version_specifier)
```